lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb

```diff
@@ -76,6 +79,13 @@ def custom_build_settings
             super.merge(settings)
           end
 
+          # @return [Boolean] whether this target requires an `Embed Frameworks` script phase
+          #
+          def embed_frameworks_script_required?
+            includes_dynamic_xcframeworks = target.xcframeworks_by_config.values.flatten.map(&:build_type).any?(&:dynamic_framework?)
```

Comment:
`flat_map(&:build_type)`?

Reply:
need `flatten` before because `values` is an array of arrays, instead of `map` returning the array

Reply:
ah yes